posthog/models/person/person.py

```diff
@@ -111,12 +107,26 @@ def split_person(self, main_distinct_id: Optional[str], max_splits: Optional[int
 
 class PersonDistinctId(models.Model):
     class Meta:
-        constraints = [models.UniqueConstraint(fields=["team", "distinct_id"], name="unique distinct_id for team")]
+        constraints = [
+            # These columns in the unique constraint are also used as the
+            # primary key columns for the ClickHouse copy of this table.
+            models.UniqueConstraint(fields=["team", "distinct_id"], name="unique distinct_id for team"),
+        ]
 
     team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE, db_index=False)
-    person: models.ForeignKey = models.ForeignKey(Person, on_delete=models.CASCADE)
     distinct_id: models.CharField = models.CharField(max_length=400)
 
+    # IMPORTANT: When reading from this table, you need to filter out rows where
+    # the person is null if you are only interested in rows that are currently
+    # associated with a valid (non-deleted) person. Rather than deleting rows
+    # from this table when a person is deleted, we keep the row around but
+    # disassociate it from the person that is being deleted. If we were to
+    # cascade person deletion to this table, the ``version`` value used by
+    # ClickHouse would appear to roll back to zero from after distinct ID reuse,
+    # leading to lost updates and data drift between the Postgres and ClickHouse
+    # copies of this table.
+    person: models.ForeignKey = models.ForeignKey(Person, on_delete=WARN_ON_CASCADE, null=True)
```

Comment:
`WARN_ON_CASCADE` is a temporary measure here, and will be eventually replaced with [`models.DO_NOTHING`](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.DO_NOTHING) which will let Postgres keep us from accidentally cascading deletes to this model in a which could lead to data drift between Postgres and ClickHouse.

In the meantime this is useful because:

1. It will help verify that we're not inadvertently triggering cascading deletes with side effects in production,
2. It keeps me from needing to update all the tests that are currently creating and deleting persons that have distinct IDs associated with them. 😛

Reply:
Wouldn't we want `on_delete` to be `models.SET_NULL` now that we support nulls? Anyways, happy we are keeping with the same behavior for now (+ logging, good idea!).